sphinx_doc/migration/Eloquent.rst

```diff
@@ -42,3 +42,17 @@ Many new behavior tree nodes were added.
 These behavior tree nodes are now BT plugins and dynamically loadable at run-time using behavior tree cpp v3.
 See ``nav2_behavior_tree`` for a full listing, or :ref:`plugins` for the current list of behavior tree plugins and their descriptions. 
 These plugins are set as default in the ``nav2_bt_navigator`` but may be overrided by the ``bt_plugins`` parameter to include your specific plugins.
+
+Map Server Re-Work
+******************
+
+Map Server now has new MapIO dynamic library. All map input-output functionality was moved from ``map_server`` and ``map_saver`` here. MapIO contains loading/saving functions working with ``OccupancyGrid`` messages: ``loadMapYaml()``, ``loadMapFromFile()``, ``loadMapFromYaml()`` and ``saveMapToFile()``. These functions could be easily called from any part of external ROS2 code even if Map Server node was not started.
+MapIO library is designed to be replaceable by a new IO library working with ``OccupancyGrid`` maps, e.g. by library with new map encoding method or any other library supporting costmaps, multifloor maps, etc...
+
+``map_saver`` was re-worked and divided into 2 parts: CLI and server. CLI part remained to be almost untouched. CLI executable was renamed from ``map_saver`` to ``map_saver_cli`` without changing its functionality. As previously, it is saving the map into a file through command-line interface. Server is a new part added to serve incoming ``save_map`` service requests. By each request it tries to listen incoming map topic, receive a message from it and write obtained map into a file.
```

Comment:
I think it would be helpful here to explain what each does. Ex. CLI is the command line tool to save 1 map once, and the server will spin in the background and can be used to save maps continuously through a service.